src/components/collapsible_nav/collapsible_nav.spec.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/// <reference types="../../../cypress/support"/>
+
+import React, { useState } from 'react';
+
+import { EuiCollapsibleNav } from './collapsible_nav';
+import { EuiHeader, EuiHeaderSectionItemButton } from '../header';
+import { EuiIcon } from '../icon';
+
+const Nav = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <EuiHeader
+      style={{ zIndex: 1001 }}
+      position="fixed"
+      sections={[
+        {
+          items: [
+            <EuiCollapsibleNav
+              style={{ top: 48 }}
+              id="navSpec"
+              isOpen={isOpen}
+              button={
+                <EuiHeaderSectionItemButton
+                  data-test-subj="navSpecButton"
+                  aria-label="Toggle main navigation"
+                  onClick={() => setIsOpen(!isOpen)}
+                >
+                  <EuiIcon type={'menu'} size="m" aria-hidden="true" />
+                </EuiHeaderSectionItemButton>
+              }
+              onClose={() => setIsOpen(false)}
+            >
+              <button data-test-subj="itemA">Item A</button>
+              <button data-test-subj="itemB">Item B</button>
+              <button data-test-subj="itemC">Item C</button>
+              <input data-test-subj="itemD" />
+            </EuiCollapsibleNav>,
+          ],
+        },
+      ]}
+    />
+  );
+};
+
+describe('EuiCollapsibleNav', () => {
+  describe('Elastic pattern', () => {
+    describe('Toggle button behavior', () => {
+      it('opens and closes nav when the main button is clicked', () => {
+        cy.mount(<Nav />);
+        cy.wait(400);
+        cy.get('[data-test-subj="navSpecButton"]')
+          .realClick()
+          .then(() => {
+            expect(cy.get('#navSpec').should('exist'));
+
+            cy.get('[data-test-subj="navSpecButton"]')
+              .realClick()
+              .then(() => {
+                expect(cy.get('#navSpec').should('not.exist'));
+              });
+          });
+      });
+
+      it('closes the nav when the overlay mask is clicked', () => {
+        cy.mount(<Nav />);
+        cy.wait(400);
+        cy.get('[data-test-subj="navSpecButton"]')
+          .realClick()
+          .then(() => {
+            cy.get('.euiOverlayMask')
+              .realClick()
+              .then(() => {
+                expect(cy.get('#navSpec').should('not.exist'));
+              });
+          });
```

Comment:
similarly, do these actually need `.then()` chaining, or can you write everything out flatly without sync issues?

```suggestion
        cy.get('[data-test-subj="navSpecButton"]').realClick();
        cy.get('.euiOverlayMask').realClick();
        expect(cy.get('#navSpec').should('not.exist'));
```

The chaining is adding a lot of verbosity that I'm suspicious might not be necessary. If I'm totally wrong and Cypress is flipping its lid without the nesting though no worries 🤷

Reply:
Looks like _some_ of the `then` chaining is required for real events, but not all. Will updated.